torchtext/datasets/text_classification.py

```diff
@@ -0,0 +1,416 @@
+import os
+import re
+import torch
+from torchtext.utils import download_from_url, extract_archive
+from torchtext.data.utils import generate_ngrams
+from torchtext.vocab import build_vocab
+import random
+from torchtext import data
+from tqdm import tqdm
+from torchtext.data import dataset
+from torchtext.data.iterator import generate_iterators
+
+
+def download(url, raw_folder, dataset_name):
+    """Download the dataset if it doesn't exist in processed_folder already."""
+
+    train_csv_path = os.path.join(raw_folder,
+                                  dataset_name + '_csv',
+                                  'train.csv')
+    test_csv_path = os.path.join(raw_folder,
+                                 dataset_name + '_csv',
+                                 'test.csv')
+    if os.path.exists(train_csv_path) and os.path.exists(test_csv_path):
+        return
+
+    os.makedirs(raw_folder)
+    filename = dataset_name + '_csv.tar.gz'
+    url = url
+    path = os.path.join(raw_folder, filename)
+    download_from_url(url, path)
+    extract_archive(path, raw_folder, remove_finished=True)
+
+    print('Dataset %s downloaded.' % dataset_name)
+
+
+def text_normalize(line):
```

Comment:
Maybe "basic_normalization_english" or such? Is there a reference here (maybe within a paper)? Can be immediately moved into c++. It's very useful in general and we can indicate that.

Reply:
the supervised learning paper didn't mention how they did pre-processing. fastText just did this pre-processing in the bash script.